src/task.ts

```diff
@@ -379,6 +379,85 @@ export class WhenAnyTask extends CompoundTask {
     }
 }
 
+/**
+ * @hidden
+ *
+ * CallHttp Task with polling logic
+ *
+ * If the HTTP requests returns a 202 status code with a 'Location' header,
+ * then a timer task is created, after which another HTTP request is made,
+ * until a different status code is returned.
+ *
+ * Any other result from the HTTP requests is the result of the whole task.
+ *
+ * The duration of the timer is specified by the 'Retry-After' header (in seconds)
+ * of the 202 response, or a default value specified by the durable extension is used.
+ *
+ */
+export class CallHttpWithPollingTask extends CompoundTask {
+    protected action: CallHttpAction;
+    private readonly defaultHttpAsyncRequestSleepDuration: moment.Duration;
+
+    public constructor(
+        id: TaskID,
+        action: CallHttpAction,
+        private readonly orchestrationContext: DurableOrchestrationContext,
+        private readonly executor: TaskOrchestrationExecutor,
+        defaultAsyncSleepRequestTime: string
+    ) {
+        super([new AtomicTask(id, action)], action);
+        this.id = id;
+        this.action = action;
+        this.defaultHttpAsyncRequestSleepDuration = moment.duration(defaultAsyncSleepRequestTime);
+    }
+
+    public trySetValue(child: TaskBase): void {
+        if (child.stateObj === TaskState.Completed) {
+            if (child.actionObj instanceof CallHttpAction) {
+                const result = child.result as DurableHttpResponse;
+                if (result.statusCode === 202 && result.headers && result.headers.Location) {
+                    const delay: moment.Duration = result.headers["Retry-After"]
+                        ? moment.duration(result.headers["Retry-After"], "s")
+                        : this.defaultHttpAsyncRequestSleepDuration;
+
+                    const currentTime = this.orchestrationContext.currentUtcDateTime;
+                    const timerFireTime = moment(currentTime).add(delay).toDate();
+
+                    // this should be safe since both types returned by this call
+                    // (DFTimerTask and LongTimerTask) are TaskBase-conforming
+                    const timerTask = (this.orchestrationContext.createTimer(
+                        timerFireTime
+                    ) as unknown) as TaskBase;
```

Comment:
I'm surprised that you need is `as unknown` cast prior to the `as TaskBase` cast, but I trust it prevents some type-checker error.

Reply:
Yep, I don't fully understand it either, but without this intermediate step you would get this error:
```
Conversion of type 'TimerTask' to type 'TaskBase' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
```